// zjuqn/classes/img.js
const fetch = require('node-fetch')

class Img {
    constructor(token) {
      if (!token) throw new Error("[zJuqn.Error] | Debes colocar el token");
      this.token = token;
    }
  
    /**
     * @param {img} img
     * @returns {img} <Buffer>
     */
     async triggered(avatar) {
      if(!avatar) throw new Error("[zJuqn.Error] | Debes colocar un avatar");
      const zjuqnapi = await fetch(`https://api.insideparadox.repl.co/api/v1/gifs/triggered?token=${this.token}&avatar=${avatar}`)
            .then(text => text.json())
            await fetch(`https://api.insideparadox.repl.co/api/v1/gifs/triggered?token=${this.token}&avatar=${avatar}`).then(errores => {
                errores.json()
                if(errores.status == "404") {
                    throw new Error("[zJuqn.Error] | Token Invalido ")
                };
                if(errores.status == "400") {
                    throw new Error("[zJuqn.Error] | El link de tu avatar es invalido ")
                }
            });
            return zjuqnapi.image//new Discord.MessageAttachment(, "zjuqnapi_triggered.gif")
  };
  async stonks(avatar) {
      if(!avatar) throw new Error("[zJuqn.Error] | Debes colocar un avatar");
      const zjuqnapi = await fetch(`https://api.insideparadox.repl.co/api/v1/img/stonks?token=${this.token}&avatar=${avatar}`)
            .then(text => text.json())
            await fetch(`https://api.insideparadox.repl.co/api/v1/img/stonks?token=${this.token}&avatar=${avatar}`).then(errores => {
                errores.json()
                if(errores.status == "404") {
                    throw new Error("[zJuqn.Error] | Token Invalido ")
                };
                if(errores.status == "400") {
                    throw new Error("[zJuqn.Error] | El link de tu avatar es invalido ")
                }
            });
            return zjuqnapi.image//new Discord.MessageAttachment(, "zjuqnapi_triggered.gif")
  };
  async nostonks(avatar) {
      if(!avatar) throw new Error("[zJuqn.Error] | Debes colocar un avatar");
      const zjuqnapi = await fetch(`https://api.insideparadox.repl.co/api/v1/img/nostonks?token=${this.token}&avatar=${avatar}`)
            .then(text => text.json())
            await fetch(`https://api.insideparadox.repl.co/api/v1/img/nostonks?token=${this.token}&avatar=${avatar}`).then(errores => {
                errores.json()
                if(errores.status == "404") {
                    throw new Error("[zJuqn.Error] | Token Invalido ")
                };
                if(errores.status == "400") {
                    throw new Error("[zJuqn.Error] | El link de tu avatar es invalido ")
                }
            });
            return zjuqnapi.image//new Discord.MessageAttachment(, "zjuqnapi_triggered.gif")
  };
  async lisa(text) {
      if(!text) throw new Error("[zJuqn.Error] | Debes colocar un texto");
      const zjuqnapi = await fetch(`https://api.insideparadox.repl.co/api/v1/img/lisa?token=${this.token}&texto=${text}`)
            .then(text => text.json())
            await fetch(`https://api.insideparadox.repl.co/api/v1/img/lisa?token=${this.token}&texto=${text}`).then(errores => {
                errores.json()
                if(errores.status == "404") {
                    throw new Error("[zJuqn.Error] | Token Invalido ")
                };
                if(errores.status == "400") {
                    throw new Error("[zJuqn.Error] | Algo salio mal porfavor intenelo mas tarde ")
                }
            });
            return zjuqnapi.image//new Discord.MessageAttachment(, "zjuqnapi_triggered.gif")
  };
  async anuncio(avatar) {
      if(!avatar) throw new Error("[zJuqn.Error] | Debes colocar un avatar");
      const zjuqnapi = await fetch(`https://api.insideparadox.repl.co/api/v1/img/anuncio?token=${this.token}&avatar=${avatar}`)
            .then(text => text.json())
            await fetch(`https://api.insideparadox.repl.co/api/v1/img/anuncio?token=${this.token}&avatar=${avatar}`).then(errores => {
                errores.json()
                if(errores.status == "404") {
                    throw new Error("[zJuqn.Error] | Token Invalido " )
                };
                if(errores.status == "400") {
                    throw new Error("[zJuqn.Error] | El link de tu avatar es invalido ")
                }
            });
            return zjuqnapi.image//new Discord.MessageAttachment(, "zjuqnapi_triggered.gif")
  };
  async gay(avatar) {
      if(!avatar) throw new Error("[zJuqn.Error] | Debes colocar un avatar");
      const zjuqnapi = await fetch(`https://api.insideparadox.repl.co/api/v1/img/gay?token=${this.token}&avatar=${avatar}`)
            .then(text => text.json())
            await fetch(`https://api.insideparadox.repl.co/api/v1/img/gay?token=${this.token}&avatar=${avatar}`).then(errores => {
                errores.json()
                if(errores.status == "404") {
                    throw new Error("[zJuqn.Error] | Token Invalido ")
                };
                if(errores.status == "400") {
                    throw new Error("[zJuqn.Error] | El link de tu avatar es invalido ")
                }
            });
            return zjuqnapi.image//new Discord.MessageAttachment(, "zjuqnapi_triggered.gif")
  };
  }

module.exports = Img;
