// zjuqn/classes/WelcomeImage.js
const Discord = require('discord.js')
const request = require('node-superfetch')
const {zJuqnError} = require('../Util/zJuqnError')

  /**
   * @param {String} options.token
   * @param {String} options.background
   * @param {String} options.avatar
   * @param {String} options.title
   * @param {String} options.color
  */

class WelcomeImage {

  constructor(options) {

        if(!options.token) throw new zJuqnError("[zJuqn.Error] | Debes ingresar un token")
        if(!options.background) throw new zJuqnError("[zJuqn.Error] | Debes ingresar el background")
        if(!options.avatar) throw new zJuqnError("[zJuqn.Error] | Debes ingresar un avatar")
        if(!options.title) throw new zJuqnError("[zJuqn.Error] | Debes agregar un titulo")
        if(!options.color) throw new zJuqnError("[zJuqn.Error] | Debes ingresar el color")

        this.token = options.token
        this.background = options.background
        this.avatar = options.avatar
        this.title = options.title 
        this.color = options.color
}
           async obtener() {
            const url = `https://api.insideparadox.repl.co/api/v1/bienvenidas?token=${this.token}&background=${this.background}&avatar=${this.avatar}&subtitulo=${this.title}&color=${this.color}`
            try {
              const { body } = await request.get(url)
            } catch (err) {
              if(err.status) {
                throw new zJuqnError(err.body.mensaje);
              }
              
            };
            
            const { body } = await request.get(url) 
        
  
          return new Discord.MessageAttachment(body.image, "zjuqnapi_welcome.png")
           
           }
}

module.exports = WelcomeImage;
